internal/serve/middleware/middleware.go

```diff
@@ -249,6 +249,28 @@ func TenantMiddleware(tenantManager tenant.ManagerInterface, authManager auth.Au
 	}
 }
 
+func BasicAuthMiddleware(adminAccount, adminApiKey string) func(http.Handler) http.Handler {
```

Comment:
I was thinking of implementing a `BasicAuthManager` that knows the `adminAccount` and `adminApiKey` and performs the validation, instead of passing two string values. 

It seemed like overkill, but if anyone thinks differently I'll switch.

Reply:
Thank you @CaioTeixeira95 ! This change from `chi` is a good one though, I added it in. https://github.com/go-chi/chi/pull/549/files

Reply:
This is great as is, good call!